Common/Source/Comm/device.cpp

```diff
@@ -674,55 +693,79 @@ PDeviceDescriptor_t devGetDeviceOnPort(int Port){
 
  // devParseStream(devIdx, c, &GPS_INFO);
 BOOL devParseStream(int portNum, char* stream, int length, NMEA_INFO *pGPS){
-  
-  PDeviceDescriptor_t d = devGetDeviceOnPort(portNum);
-  if (d && d->ParseStream) {
-    d->HB=LKHearthBeats;
-    if (d->ParseStream(d, stream, length, pGPS)) {
-          
-    }
-    return(TRUE);        
-  }
+bool  ret = FALSE;
+PDeviceDescriptor_t din = devGetDeviceOnPort(portNum);
+PDeviceDescriptor_t d = NULL;
 
-  return(FALSE);    
+    for(int dev =0; dev < NUMDEV; dev++)
+    {
+      d = &DeviceList[dev];
+      if (din && d && d->ParseStream)
+      {
+       if((d->iSharedPort == portNum) ||  (d->Port == portNum))
+       {
+         d->HB=LKHearthBeats;
+         if (d->ParseStream(din, stream, length, pGPS)) {
+          
+         }
+         ret =TRUE;
+       }
+     }
+   }
+  return(ret);
 }    
 
 
 
 // Called from Port task, after assembly of a string from serial port, ending with a LF
 BOOL devParseNMEA(int portNum, TCHAR *String, NMEA_INFO *pGPS){
-
+bool  ret = FALSE;
   LogNMEA(String, portNum); // We must manage EnableLogNMEA internally from LogNMEA
 
   PDeviceDescriptor_t d = devGetDeviceOnPort(portNum);
+  PDeviceDescriptor_t d2;
   // intercept device specific parser routines 
   if (d != NULL){
-	d->HB=LKHearthBeats;
+    d->HB=LKHearthBeats;
 
-    if (d->pDevPipeTo && d->pDevPipeTo->Com) {
-	// stream pipe, pass nmea to other device (NmeaOut)
-	// TODO code: check TX buffer usage and skip it if buffer is full (outbaudrate < inbaudrate)
-	d->pDevPipeTo->Com->WriteString(String);
-    }
-#ifdef RADIO_ACTIVE    
-    if (devDriverActivated(TEXT("PVCOM")))
-      PVCOMParseString(d, String, pGPS);
-#endif
-    if (d->ParseNMEA != NULL)
-	if ((d->ParseNMEA)(d, String, pGPS)) {
+    for(int dev =0; dev < NUMDEV; dev++)
+    {
+      d2 = &DeviceList[dev];
+      if (d2)
+       if((d2->iSharedPort == portNum) ||  (d2->Port == portNum))
+         if (d2->ParseNMEA != NULL)
+	   if ((d2->ParseNMEA)(d, String, pGPS)) {
 		//GPSCONNECT  = TRUE; // NO! 121126
-		return(TRUE);
-	}
+	      ret = TRUE;
+	  }
+    }
   }
 
+
   if(String[0]=='$') {  // Additional "if" to find GPS strings
 	if(d->nmeaParser.ParseNMEAString_Internal(String, pGPS)) {
```

Comment:
only for master Port ?
if yes, we need if string are already used by Driver::ParseNMEA